streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -294,13 +305,27 @@ public void shouldIterateOverRange() {
         assertEquals(items, results.size());
     }
 
+    @Test
+    public void shouldReverseIterateOverRange() {
+        final int items = addItemsToCache();
+        final KeyValueIterator<Bytes, byte[]> range =
+            store.reverseRange(bytesKey(String.valueOf(0)), bytesKey(String.valueOf(items)));
+        final List<Bytes> results = new ArrayList<>();
+        while (range.hasNext()) {
+            results.add(range.next().key);
+        }
+        assertEquals(items, results.size());
```

Comment:
Can we add some tests that verify the actual contents + order of the reverse range?

Reply:
ack. added.